mlflow/gateway/config.py

```diff
@@ -345,10 +345,20 @@ class Config:
         }
 
 
+class Limit(LimitModel):
+    calls: int
+    key: Optional[str] = None
+    renewal_period: str
+
+
 class GatewayConfig(ConfigModel):
     routes: List[RouteConfig]
 
 
+class LimitsConfig(ConfigModel):
+    limits: Optional[List[Limit]] = []
```

Comment:
```suggestion
    limits: List[Limit] = []
```

Do we need `Optional`?

Reply:
Actually there is an issue we are investigating: GetLimits rpc returns {} if there is no limit, not sure if it should be {"limits": []} instead, seems to be a proto2 thing, still investigating whether we should enforce {"limits": []},.

Reply:
Sounds like a proto2 thing, then `Optional` is ok.

Reply:
confirmed with @dbczumar that we will be aligned with SearchRoutes behavior: return `{}` hence I think limits should be Optional otherwise it cannot parse the `{}` response.

Reply:
Sounds great!